src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_internal.h

```diff
@@ -73,8 +73,8 @@ U_CAPI int32_t U_EXPORT2 ucal_getWindowsTimeZoneID(const UChar* id, int32_t len,
     PER_FUNCTION_BLOCK(u_tolower, libicuuc, true) \
     PER_FUNCTION_BLOCK(u_toupper, libicuuc, true) \
     PER_FUNCTION_BLOCK(u_uastrncpy, libicuuc, true) \
-    PER_FUNCTION_BLOCK(ubrk_close, libicui18n, true) \
-    PER_FUNCTION_BLOCK(ubrk_openRules, libicui18n, true) \
+    PER_FUNCTION_BLOCK(ubrk_close, libicuuc, true) \
```

Comment:
Is there a rule/guidance on when to use `libicuuc` and when to use `libicui18n`?

Reply:
Usually, is to find it in the exports of the library. for this specific case, it happened that the API caused the issue `ubrk_close` is exported from both of them.

```
nm -D /usr/lib/x86_64-linux-gnu/libicui18n.so.66 | grep -i ubrk_
                 U ubrk_close_66

nm -D /usr/lib/x86_64-linux-gnu/libicuuc.so.66.1 | grep -i ubrk_
00000000001206a0 T ubrk_close_66
```

But looking at https://docs.microsoft.com/en-us/uwp/win32-and-com/win32-extension-apis I am seeing Windows define such APIs in `icuuc` library.

Reply:
But what's the general rule for when we use one over the other? Why do some of the functions here use `libicuuc` and some use `libicui18n`? Is it "unicode" functions vs. "everything else"?

Reply:
Not related to Unicode. It is just a matter knowing the API is exported from which module from the text section. like:

```
nm -D /usr/lib/x86_64-linux-gnu/libicuuc.so.66.1 | grep -i ubrk_
00000000001206a0 T ubrk_close_66
```

Notice the 'T' which indicating the text section. 
When I used `libicui18n`in the first place, it was a mistake which happened to work. There is no 2 sets of the ICU APIs. it is just one set. It happened that libicui18n included the symbol for the API which get forwarded to the actual implementation (in `icuuc`). In ICU NuGet version, `libicui18n` didn't include this export.